### ISSUE TITLE
High resolution problem

### ISSUE THREAD

**Issue by Destiny-Dream1** [Jamy-L/Handheld-Multi-Frame-Super-Resolution#31]
Hello, I have a question to ask. I have downloaded the Samsung dataset and obtained low-resolution images, and the program runs successfully. However, I have been searching for a long time but cannot find the high-resolution images to compare the results and calculate PSNR and SSIM. I appreciate any help.

Thank you so much.

**Comment by teboli** [comment]
The samsung images come from the burst SR dataset https://github.com/goutamgmb/deep-burst-sr (paper explaining the collection procedure: https://arxiv.org/abs/2101.10997). Please have a look at the full-resolution dataset section to get the images, there should be a Canon folder with the targets images in it for each sample. Here is a sample of the dataset:

<img width="747" alt="Screenshot 2023-12-09 at 12 29 20" src="https://github.com/Jamy-L/Handheld-Multi-Frame-Super-Resolution/assets/19273558/4aa6dbc9-a5e0-4323-aeff-9ca3ff7718d6">

The goal of the authors is to learn 4x SR with DSLR high-resolution sRGB photographs as targets. They thus learn in the process the ISP for rendering a Samsung G8 raw images into a Canon 5DIV sRGB rendering, including sharpening. The targets are the canon's JPEG companion images. However a fundamental difference between deep learning approaches trained with such targets and the handheld method is that the former are pattern matching techniques that may hallucinate details whereas the latter aims solely at interpolating a good denoised and dealiased image from the noisy and aliased samples alone. Furthermore, the authors of the original paper noticed that you can really obtained new details up to 2x, beyond this point you are limited by the optics! This reasoning was made for the Google Pixel 3 phone but you can assume that the contemporary Samsung G8 model has a similar lens, and thus yields similarly aliased raw images.

As a result, the output of the handheld approach comes without the whole post-processing of any camera ISP. It is up to you to apply any post-processing you like, and it must include deconvolution/sharpening (please have a look at the explanations of papers in the same vein like: https://people.duke.edu/~sf59/SRfinal.pdf). Also, the target Canon photo and the reference frame of the burst are only weakly aligned. If you really want to measure a pixel-to-pixel loss such as the PSNR or the SSIM, you may want to use the CoBI loss from this project: https://ceciliavision.github.io/project-pages/zoom.html Note that it is beyond the scope of this repo that focused on reproducing as faithfully as possible the original handheld paper!

tl;dr: You can increase by 4x the resolution of the final image to match that of the targets of the burst SR dataset, but this comparison is comparing apples and oranges because you have to apply your own ISP pipeline to render the final sRGB image (since Canon's ISP is a black-box function, it is very unlikely that you will have a perfect 1:1 reproduction), and the regression techniques are different: pattern matching (deep learning) v. interpolation (handheld, kernel regression). 

Let me know if you still have questions :)

Best,

Thomas

**Comment by Destiny-Dream1** [comment]
Hello, I've understood the method for obtaining the dataset, and I'm currently studying both the program and the paper. However, what I would like to inquire about is conducting PSNR and SSIM tests and comparisons between your super-resolution method and mine. I'm wondering if there's a faster way to obtain high-resolution results. Thank you~~
